digdag-standards/src/main/java/io/digdag/standards/command/EcsCommandExecutor.java

```diff
@@ -659,7 +661,9 @@ protected void setEcsContainerOverrideCommand(
             bashArguments.add(s("popd"));
         }
         bashArguments.add(s("tar -zcf %s  --exclude %s --exclude %s .digdag/tmp/", outputProjectArchivePathName, relativeProjectArchivePath.toString(), outputProjectArchivePathName));
-        bashArguments.add(s("curl -s -X PUT -T %s -L \"%s\"", outputProjectArchivePathName, outputProjectArchiveDirectUploadUrl));
+        // retry by exponential backoff 1+2+4+8+16+32+64=127 seconds
+        // Note that it's intended to curl exit 0 on http errors since it's only for LogWatch logging
+        bashArguments.add(s("curl --retry 7 --retry-connrefused -s -X PUT -T %s -L \"%s\"", outputProjectArchivePathName, outputProjectArchiveDirectUploadUrl));
```

Comment:
* isn't `--fail` option unnecessary? digdag server maintainer and workflow user are different. basically the server maintainer doesn't want to include error message details like S3 file path or pre-signed URL to the user.
* what do you think we could extract the retry count as parameters? because that's depending on the use case. probably some cases, server maintainer don't want to give up the retry because the script command succeeds but the workflow task fails.

Reply:
@muga 

> isn't --fail option unnecessary?  the server maintainer doesn't want to include error message details like S3 file path or pre-signed URL to the user.

Good point.

`-fail` command is introduced for retrying by Digdag operator level for better failure analysis. 

Without `-fail`, the curl command exit 0 for http transient errors and the following [tar command](https://github.com/treasure-data/digdag/pull/1436/files/42f762054d33efd40b6f10cf85a801458c823ad3#diff-bede7c26064840fb802ecba2264858dcR651) fails. Then, s3 file path leaks.

```
tar -zxf notfound
tar: Error opening archive: Failed to open 'notfound'
```

Since `-s (--silent)` option is specified in curl commands, no error message is going to be shown when exit 22 happens. So, no s3 path leak with --fail option. But, leak can be happened without --fail.

So,`--fail` is necessary.

> what do you think we could extract the retry count as parameters? 

I think it's not required because miss configuration (e.g., infinite retry) can make digdag server unstable.
Only configurable by system administrator (who runs digdag server) is preferred.

Reply:
> Only configurable by system administrator (who runs digdag server) is preferred.

yes, so the retry count needs to be extracted and parameterized as config param. otherwise it's not easy for the system administrator (meant digdag server maintainer in my wording) to configure it.

Reply:
@muga 

Introduced `agent.command_executor.ecs.retry_task_scripts_downloads` and `agent.command_executor.ecs.retry_task_output_uploads`. 

`--retry 0` makes curl do no retries. See [man](https://curl.haxx.se/docs/manpage.html).

Reply:
nice, thanks! I missed one more thing for 1st one.

> So, I think--fail is preferred.

agree. please add `--fail` option.

Reply:
Added `agent.command_executor.ecs.disable_curl_fail_opt=true` .